### ISSUE TITLE
Add logic to build report

### ISSUE THREAD

**Issue by epijim** [insightsengineering/r-pkg-installation-validation#9]
Add logic to build report - e.g. modify this file: https://github.com/insightsengineering/r-pkg-installation-validation/blob/main/report-generator.R

**Comment by epijim** [comment]
@dinakar29 and @dgkf - I'm def a newbie on gh-actions, as I tended to use `usethis` actions in the past. I got very confused by the way environments work when sourcing in a gh-action. e.g. this uses rocker, which I don't think comes with covr. I guess that means we need to modify this action (as editing the gh-action on the target repo to install covr in an earlier step doesn't seem to carry over). 

Does that also mean we need to have in this action to install package dependencies (so unit tests run)?

**Comment by epijim** [comment]
@dinakar29 - I'm not sure this line on the Dockerfile works: https://github.com/insightsengineering/r-pkg-installation-validation/blob/add/generator/Dockerfile#L8

As when I source the action I get it can't find the template: https://github.com/openpharma/visR/runs/3544751668?check_suite_focus=true#step:4:246

If I manually add a template - it works e.g. https://github.com/openpharma/visR/actions/runs/1213152385

**Comment by epijim** [comment]
Note I ran into this issue as a blocked when trying it out on `rbmi`: https://github.com/Genentech/covtracer/issues/4

**Comment by dgkf** [comment]
@epijim - I updated `covtracer` to safely handle this scenario and at least proceed using `NA` srcrefs. Even if the result of `test_trace_df` doesn't include test srcrefs, does it at least proceed without errors now? 

Just for awareness, the "tracing" behavior isn't contingent on having test `srcrefs` - they're just nice to have for more definitive sources of test code. We can still pull the test descriptions even without the srcrefs, but we might end up in situations where multiple tests have the same description and we can't distinguish which one links to which. Perhaps from a documentation perspective this is irrelevant.

I'll continue to explore this. My guess is there's some environment flag or compiled default behavior that needs to be set in the environment before `covr` will install the package to a temporary location and parse files while preserving their source code.

**Comment by dgkf** [comment]
Thanks for flagging this @epijim - Interesting. I was hoping this was behind us. I believe this boils down to not having an installed version of the package with srcrefs (installed with `--with-keep.source`).

I'll have to brush up on the action and try to retrace its steps to figure out if the package isn't getting installed with this extra package data somehow.

**Comment by dgkf-roche** [comment]
I've been able to reproduce the issue. Will follow up once I've got a fix pulled in.
